### ISSUE TITLE
Argument reflection for task

### ISSUE THREAD

**Issue by basuke** [climphp/Clim#3]
Automated argument passing from container is a fancy feature for cli app.

example:
```php
$app->task(function(\PDO $db, array $args) {
    // $db will be pulled from container
    // equivalent to
    // /** @var \PDO $db */
    // $db = $this->db;
});

**Comment by basuke** [comment]
For $arguments and $options which is current arguments for task, there must be a special rule for them.

- `$arguments` and `$args` are reserved argument for `$arguments`
- `$options` and `$opts` are reserved argument for `$options`
- It's possible to specify those should be array or Hash or whatever classes accept array to its constructor. The default is `array`.

**Comment by basuke** [comment]
Implemented on `v1.5.0`
